### ISSUE TITLE
Decrease size of Docker image

### ISSUE THREAD

**Issue by dector** [exercism/kotlin-test-runner#5]
@exercism/ops please check [Dockerfile](https://github.com/exercism/kotlin-test-runner/blob/master/Dockerfile).

Optimization advises are very welcomed!

**Comment by ccare** [comment]
Hello @dector 

I can't comment much on size, because I haven't built it myself yet. Although it looks like you can't make it _that_ much slimmer without changing the base image.

One thing you'll need to tweak though is to setup env-vars in your run.sh file. We're not using the docker as the container runtime, so we won't have the config expressed in the ENV commands.

e.g.

To honour the `ENV` here: https://github.com/exercism/kotlin-test-runner/blob/master/Dockerfile#L39 You'll probably want to add that env-var to the run.sh script

Any ENV declarations from the Dockerfile inheritance chain will also be missing. This often doesn't matter too much, as not all are needed. However, I've noted a line that will probably be necessary. See - https://github.com/AdoptOpenJDK/openjdk-docker/blob/master/11/jdk/alpine/Dockerfile.hotspot.releases.slim#L95

I'd suggest adding 

```
export JAVA_HOME=/opt/java/openjdk
export PATH="/opt/java/openjdk/bin:$PATH"
export PATH="${PATH}:/home/gradle/bin"
```

to the run.sh file

**Comment by dector** [comment]
@ccare Thanks for your advices. Will take a look at it.

**Comment by ErikSchierboom** [comment]
@dector I took a quick look at the Dockerfile, and it looks very good to me. I'm no Java expert by any means, but it looks like you're copying the bare minimum to run the test runner, as well as using Alpine as the base image. That looks fairly optimal.

**Comment by devsnice** [comment]
I updated the maven image. It allows to decrease the result image from 730 to 540mb,  [PR](https://github.com/exercism/kotlin-test-runner/pull/43)

In general I haven't got the idea with gradle + multiple maven runs, what was the problem to run all tests as part of the gradle build process? There can be multiple gradle modules for this purpose

**Comment by ErikSchierboom** [comment]
> In general I haven't got the idea with gradle + multiple maven runs, what was the problem to run all tests as part of the gradle build process? There can be multiple gradle modules for this purpose

The problem was that gradle was incredibly slow. See https://github.com/exercism/kotlin-test-runner/pull/34#issuecomment-971641689

**Comment by ErikSchierboom** [comment]
Closed by #43
